openlibrary/plugins/worksearch/code.py

```diff
@@ -470,9 +468,18 @@ def isbn_redirect(self, isbn_param):
         if len(editions):
             raise web.seeother(editions[0])
 
-    def GET(self):
+    def GET(self, subject=None):
         global ftoken_db
         i = web.input(author_key=[], language=[], first_publish_year=[], publisher_facet=[], subject_facet=[], person_facet=[], place_facet=[], time_facet=[], public_scan_b=[])
+        if subject:
+            _facet = subject['subject_type']
+            _subj = unicode(subject['name'])
```

Comment:
```suggestion
            _subj = six.text_type(subject['name'])
```
https://six.readthedocs.io/index.html#six.text_type

If `subject['name']` was coming in at bytes, you would want to `subject['name'].decode(‘utf-8')` instead.